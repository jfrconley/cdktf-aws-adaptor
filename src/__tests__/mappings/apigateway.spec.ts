import { ApiGatewayAccount } from "@cdktf/provider-aws/lib/api-gateway-account/index.js";
import { ApiGatewayBasePathMapping } from "@cdktf/provider-aws/lib/api-gateway-base-path-mapping/index.js";
import { CloudcontrolapiResource } from "@cdktf/provider-aws/lib/cloudcontrolapi-resource/index.js";
import { CfnAccount, CfnBasePathMapping, CfnDeployment, CfnStage } from "aws-cdk-lib/aws-apigateway";
import { setupJest } from "cdktf/lib/testing/adapters/jest.js";
import { registerMappings } from "../../mappings/index.js";
import { synthesizeElementAndTestStability } from "../helpers.js";

setupJest();
registerMappings();

describe("Apigateway mappings", () => {
  it("should map AWS::ApiGateway::Account", () => {
    synthesizeElementAndTestStability(
      CfnAccount,
      {
        cloudWatchRoleArn: "cw-role-arn",
      },
      ApiGatewayAccount,
      {
        cloudwatchRoleArn: "cw-role-arn",
      },
    );
  });

  it("should map AWS::ApiGateway::Deployment", () => {
    synthesizeElementAndTestStability(
      CfnDeployment,
      {
        restApiId: "rest-api-id",
        stageName: "stage-name",
        description: "description",
        stageDescription: {
          creationStack: ["stack"],
          cacheClusterEnabled: true,
          cacheClusterSize: "cache-cluster-size",
          canarySetting: {
            percentTraffic: 50,
            stageVariableOverrides: {
              "stage-variable": "stage-variable-value",
            },
          },
          accessLogSetting: {
            destinationArn: "destination-arn",
          },
          cacheDataEncrypted: true,
          cacheTtlInSeconds: 60,
          cachingEnabled: true,
          clientCertificateId: "client-certificate-id",
          dataTraceEnabled: true,
          loggingLevel: "logging-level",
          metricsEnabled: true,
          documentationVersion: "documentation-version",
          tracingEnabled: true,
          methodSettings: [
            {
              dataTraceEnabled: true,
              httpMethod: "http-method",
              loggingLevel: "logging-level",
              metricsEnabled: true,
              resourcePath: "resource-path",
              throttlingBurstLimit: 60,
              throttlingRateLimit: 60,
              cachingEnabled: true,
              cacheDataEncrypted: true,
              cacheTtlInSeconds: 60,
            },
          ],
          throttlingBurstLimit: 60,
          throttlingRateLimit: 60,
        },
        deploymentCanarySettings: {
          percentTraffic: 50,
          stageVariableOverrides: {
            "stage-variable": "stage-variable-value",
          },
          creationStack: ["stack"],
          useStageCache: true,
        },
      },
      CloudcontrolapiResource,
      {
        typeName: "AWS::ApiGateway::Deployment",
        desiredState: expect.stringContaining(""),
      },
    );
  });

  it("should map AWS::ApiGateway::Stage", () => {
    synthesizeElementAndTestStability(
      CfnStage,
      {
        accessLogSetting: {
          destinationArn: "destination-arn",
        },
        cacheClusterEnabled: true,
        cacheClusterSize: "cache-cluster-size",
        canarySetting: {
          percentTraffic: 50,
          stageVariableOverrides: {
            "stage-variable": "stage-variable-value",
          },
        },
        clientCertificateId: "client-certificate-id",
        stageName: "stage-name",
        description: "description",
        documentationVersion: "documentation-version",
        tracingEnabled: true,
        methodSettings: [],
        restApiId: "rest-api-id",
        deploymentId: "deployment-id",
      },
      CloudcontrolapiResource,
      {
        typeName: "AWS::ApiGateway::Stage",
        desiredState: expect.stringContaining(""),
      },
    );
  });

  it("should map AWS::ApiGateway::BasePathMapping", () => {
    synthesizeElementAndTestStability(
      CfnBasePathMapping,
      {
        restApiId: "rest-api-id",
        basePath: "base-path",
        domainName: "domain-name",
        stage: "stage",
        id: "id",
      },
      ApiGatewayBasePathMapping,
      {
        apiId: "rest-api-id",
        basePath: "base-path",
        domainName: "domain-name",
        stageName: "stage",
        id: "id",
      },
    );
  });
});
