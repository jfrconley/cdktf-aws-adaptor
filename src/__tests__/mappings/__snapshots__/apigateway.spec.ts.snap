// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Apigateway mappings > should map AWS::ApiGateway::Account 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_api_gateway_account\\": {
      \\"resource_22C949BF\\": {
        \\"cloudwatch_role_arn\\": \\"cw-role-arn\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.22.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_api_gateway_account.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::BasePathMapping 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_api_gateway_base_path_mapping\\": {
      \\"resource_22C949BF\\": {
        \\"api_id\\": \\"rest-api-id\\",
        \\"base_path\\": \\"base-path\\",
        \\"domain_name\\": \\"domain-name\\",
        \\"id\\": \\"id\\",
        \\"stage_name\\": \\"stage\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.22.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::Deployment 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudcontrolapi_resource\\": {
      \\"resource_22C949BF\\": {
        \\"desired_state\\": \\"{\\\\\\"DeploymentCanarySettings\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"},\\\\\\"UseStageCache\\\\\\":true},\\\\\\"Description\\\\\\":\\\\\\"description\\\\\\",\\\\\\"RestApiId\\\\\\":\\\\\\"rest-api-id\\\\\\",\\\\\\"StageDescription\\\\\\":{\\\\\\"AccessLogSetting\\\\\\":{\\\\\\"DestinationArn\\\\\\":\\\\\\"destination-arn\\\\\\"},\\\\\\"CacheClusterEnabled\\\\\\":true,\\\\\\"CacheClusterSize\\\\\\":\\\\\\"cache-cluster-size\\\\\\",\\\\\\"CacheDataEncrypted\\\\\\":true,\\\\\\"CacheTtlInSeconds\\\\\\":60,\\\\\\"CachingEnabled\\\\\\":true,\\\\\\"CanarySetting\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"}},\\\\\\"ClientCertificateId\\\\\\":\\\\\\"client-certificate-id\\\\\\",\\\\\\"DataTraceEnabled\\\\\\":true,\\\\\\"DocumentationVersion\\\\\\":\\\\\\"documentation-version\\\\\\",\\\\\\"LoggingLevel\\\\\\":\\\\\\"logging-level\\\\\\",\\\\\\"MethodSettings\\\\\\":[{\\\\\\"CacheDataEncrypted\\\\\\":true,\\\\\\"CacheTtlInSeconds\\\\\\":60,\\\\\\"CachingEnabled\\\\\\":true,\\\\\\"DataTraceEnabled\\\\\\":true,\\\\\\"HttpMethod\\\\\\":\\\\\\"http-method\\\\\\",\\\\\\"LoggingLevel\\\\\\":\\\\\\"logging-level\\\\\\",\\\\\\"MetricsEnabled\\\\\\":true,\\\\\\"ResourcePath\\\\\\":\\\\\\"resource-path\\\\\\",\\\\\\"ThrottlingBurstLimit\\\\\\":60,\\\\\\"ThrottlingRateLimit\\\\\\":60}],\\\\\\"MetricsEnabled\\\\\\":true,\\\\\\"ThrottlingBurstLimit\\\\\\":60,\\\\\\"ThrottlingRateLimit\\\\\\":60,\\\\\\"TracingEnabled\\\\\\":true},\\\\\\"StageName\\\\\\":\\\\\\"stage-name\\\\\\"}\\",
        \\"type_name\\": \\"AWS::ApiGateway::Deployment\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.22.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${jsondecode(aws_cloudcontrolapi_resource.resource_22C949BF.properties).DeploymentId}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`Apigateway mappings > should map AWS::ApiGateway::Stage 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudcontrolapi_resource\\": {
      \\"resource_22C949BF\\": {
        \\"desired_state\\": \\"{\\\\\\"AccessLogSetting\\\\\\":{\\\\\\"DestinationArn\\\\\\":\\\\\\"destination-arn\\\\\\"},\\\\\\"CacheClusterEnabled\\\\\\":true,\\\\\\"CacheClusterSize\\\\\\":\\\\\\"cache-cluster-size\\\\\\",\\\\\\"CanarySetting\\\\\\":{\\\\\\"PercentTraffic\\\\\\":50,\\\\\\"StageVariableOverrides\\\\\\":{\\\\\\"stage-variable\\\\\\":\\\\\\"stage-variable-value\\\\\\"}},\\\\\\"ClientCertificateId\\\\\\":\\\\\\"client-certificate-id\\\\\\",\\\\\\"DeploymentId\\\\\\":\\\\\\"deployment-id\\\\\\",\\\\\\"Description\\\\\\":\\\\\\"description\\\\\\",\\\\\\"DocumentationVersion\\\\\\":\\\\\\"documentation-version\\\\\\",\\\\\\"MethodSettings\\\\\\":[],\\\\\\"RestApiId\\\\\\":\\\\\\"rest-api-id\\\\\\",\\\\\\"StageName\\\\\\":\\\\\\"stage-name\\\\\\",\\\\\\"TracingEnabled\\\\\\":true}\\",
        \\"type_name\\": \\"AWS::ApiGateway::Stage\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.22.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
